### PR TITLE
Bugfix/sim 1211/properly sum magnitudes

### PR DESCRIPTION
**Pull request by danielsf**
I discovered a bug in the sum_magnitudes method of PhotometryGalaxies.  If any one component (bulge, disk or agn) had NaN as its magnitude (because the galaxy did not have an SED associated with that component), the total magnitude for the galaxy was returned as NaN.

I have fixed that problem and added unit tests to make sure that sum_magnitudes works as expected.

This should be a quick review.  I would like to be able to incorporate this into an lsst_sims release tomorrow (Friday 6/19) evening.  If you can't look at it by then, it's not a big deal.  But I would really appreciate it if you could.

Thanks a ton,

Scott

### CODE REVIEW

File: python/lsst/sims/photUtils/Photometry.py [hunk 1]
```diff
@@ -569,7 +569,8 @@ def sum_magnitudes(self, disk = None, bulge = None, agn = None):
             raise RuntimeError("Arguments of sum_magnitudes need to be " +
```
Comment (jbkalmbach) [inline python/lsst/sims/photUtils/Photometry.py]: I didn't know where else to put this, but should that say "same type" above on lines 553 & 562?

File: python/lsst/sims/photUtils/Photometry.py [hunk 2]
```diff
@@ -569,7 +569,8 @@ def sum_magnitudes(self, disk = None, bulge = None, agn = None):
...
                                "either floats or numpy arrays; you appear to have passed %s " % baselineType)
 
-        mm_o = 22.
+        mm_0 = 22.
```
Comment (jbkalmbach) [inline python/lsst/sims/photUtils/Photometry.py]: Can you document where this zero-point magnitude value comes from? According to this: https://www.sdss3.org/dr8/algorithms/magnitudes.php I see this should maybe be 22.5?

File: tests/testPhotometry.py [hunk 3]
```diff
@@ -131,6 +131,130 @@ def testGalaxies(self):
...
+        answer[5] = -2.5*numpy.log10(bulge_flux[5]) + mm_0
+        answer[6] = -2.5*numpy.log10(bulge_flux[6]+disk_flux[6]+agn_flux[6]) + mm_0
+        answer[7] = numpy.NaN
+
+        phot = PhotometryGalaxies()
+        test = phot.sum_magnitudes(bulge=bulge, disk=disk, agn=agn)
+
+        numpy.testing.assert_array_almost_equal(test, answer, decimal=10)
+
+        for ix, (bb, dd, aa, truth) in enumerate(zip(bulge, disk, agn, answer)):
```
Comment (jbkalmbach) [inline tests/testPhotometry.py]: I really like this construction. I'm going to have to start using it.

**Comment by jbkalmbach** [comment]
Approved.

File: python/lsst/sims/photUtils/Photometry.py (hunk 2 above)
Comment (danielsf) [inline python/lsst/sims/photUtils/Photometry.py]: I'm actually not sure why it is 22 (and, in fact, it is not 22 in the Sed.calcMag methods).  For this purpose, however, it should not matter, right?  I'm converting magnitudes into fluxes, renormalizing the fluxes by 10^(-0.4*mm_0), summing the fluxes, and then dividing out the normalizing factor when I convert back into magnitudes.

File: python/lsst/sims/photUtils/Photometry.py (hunk 1 above)
Comment (danielsf) [inline python/lsst/sims/photUtils/Photometry.py]: why on Earth would we want our error messages to make sense...

good catch

**Pull request merged**